src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -121,7 +121,7 @@ class LedgerImpl(
       block: Block
   )(implicit blockExecutionScheduler: Scheduler): Task[BlockImportResult] = {
     val bestBlockHash = blockchain.getBestBlockHash()
-    blockchain.getChainWeightByHash(bestBlockHash) match {
+    blockchain.getChainWeightByHash(block.header.parentHash) match {
```

Comment:
is omitting the current block a desired step?

Reply:
the current block is being processed, and the weight will be increased in BlockExecution based on what we pass as parentWeight

Reply:
so this is a straight up bug?

Reply:
no, in the previous case, importToTop means that currentBestBlock is the parent. And in case of reorganisation, we have to keep the weight of the bestBlock around, so we can check if another branch has a higher wieght